src/aiai_eval/local_model_utils.py

```diff
@@ -0,0 +1,375 @@
+            model_folder=model_folder,
+            user_prompt="Please specify the Hugging Face ID of the tokenizer to use: ",
+        ),
+        revision="",
+        framework=Framework.PYTORCH,
+        id2label=get_from_config(
+            key="id2label",
+            expected_type=list,
+            model_folder=model_folder,
+            user_prompt="Please specify the labels in the order the model was trained "
+            "(comma-separated), or press enter to use the default values "
+            f"[{', '.join(dataset_id2label)}]: ",
+            user_prompt_default_value=dataset_id2label,
+        ),
+    )
+
+
+def get_from_config(
+    key: str,
+    expected_type: Type,
+    model_folder: Union[str, Path],
+    default_value: Optional[Any] = None,
+    user_prompt: Optional[str] = None,
+    user_prompt_default_value: Optional[Any] = None,
+) -> Any:
+    """Get an attribute from the local model configuration.
+
+    If the attribute is not found in the local model configuration, then the user
+    will be prompted to enter it, after which it will be saved to the local model
+    configuration. If the configuration file does not exist, then a new one will be
+    created named `config.json`.
+
+    Args:
+        key (str):
+            The key to get from the configuration.
+        expected_type (Type):
+            The expected type of the value.
+        model_folder (str or Path):
+            Path to the model folder.
+        default_value (Any or None, optional):
+            The default value to use if the attribute is not found in the local model
+            configuration. If None then the user will be prompted to enter the value.
+            Defaults to None.
+        user_prompt (str or None, optional):
+            The prompt to show the user when asking for the value. If None then the
+            prompt will be automatically generated. Defaults to None.
+        user_prompt_default_value (Any or None, optional):
+            The default value that a user can press Enter to use, when prompted. If
+            None then the user cannot choose a default value. Defaults to None.
+
+    Returns:
+        Any:
+            The value of the key, of data type `expected_type`.
+    """
+    # Ensure that the model_folder is a Path object
+    model_folder = Path(model_folder)
+
+    # Get the candidate configuration files
+    config_paths = list(model_folder.glob("*.json"))
+
+    # If there isn't a config then we set it to a blank dictionary. Otherwise, we load
+    # the config file
+    if not config_paths:
+        config_path = model_folder / "config.json"
+        config = dict()
+    else:
+        config_path = config_paths[0]
+        config = json.loads(config_path.read_text())
+
+    # If the key is not in the config then we either use the default value or prompt
+    # the user to enter it
+    if key not in config:
+
+        # If the default value is set and is of the correct type, then we use it
```

Comment:
Maybe wrap the contents of this loop in a while loop, and break it once we have received valid input, something like:

```
for key not in config:
    user_gave_invalid_value = True
    while user_gave_invalid_value:
        ...
        [Do bunch of stuff]
        ...
        try:
            else:
                if expected_type is str:
                    config[key] = user_input
                elif expected_type is int:
                    config[key] = int(user_input)
                elif expected_type is float:
                    config[key] = float(user_input)
                elif expected_type is bool:
                    config[key] = user_input.lower() == "true"
                elif expected_type is list:
                    config[key] = user_input.split()
                elif expected_type is dict:
                    config[key] = dict(
                        item.split("=") for item in user_input.split(",")
                    )
                user_gave_invalid_value = False
        except ValueError as e:
            [Tell people "hallo" is not an valid int value... and they should try again]
```

Right now we are assuming users can provide correct input, which is not true :(

Reply:
I just realized i am mixing try/except and a if/else blocks in the above, but you get my gist :)

Reply:
Makes sense! Coming up 🙂